src/App.tsx

```diff
@@ -1,61 +1,123 @@
 import './App.scss';
+import { ChangeEvent, FormEvent, useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const findUserById = (userId: number) => {
+  return usersFromServer.find(usr => usr.id === userId);
+};
+
+const getPreparedTodos = () => {
+  return todosFromServer.map(todo => {
+    return {
+      ...todo,
+      user: findUserById(todo.userId) || null,
+    };
+  });
+};
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(getPreparedTodos());
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [title, setTitle] = useState('');
+  const [titleInputError, setTitleInputError] = useState(false);
+  const [userSelectError, setUserSelectError] = useState(false);
+
+  const reset = () => {
+    setTitle('');
+    setSelectedUserId(0);
+    setTitleInputError(false);
+    setUserSelectError(false);
+  };
+
+  function handleSubmit(event: FormEvent<HTMLFormElement>): void {
+    event.preventDefault();
+
+    setTitleInputError(!title);
+    setUserSelectError(!selectedUserId);
+
+    if (!title || !selectedUserId) {
+      return;
+    }
+
+    setTodos(currentTodos => {
+      const newId = Math.max(...currentTodos.map(todo => todo.id)) + 1;
```

Comment:
The calculation of `newId` assumes that `currentTodos` is not empty. If `currentTodos` is empty, `Math.max` will return `-Infinity`, which will cause `newId` to be `NaN`. Consider handling the case where `currentTodos` is empty to avoid this issue.